qiskit_nature/problems/second_quantization/lattice/lattices/lattice.py

```diff
@@ -110,15 +117,20 @@ class LatticeDrawStyle:
 class Lattice:
     """General Lattice."""
 
-    def __init__(self, graph: PyGraph) -> None:
+    def __init__(self, graph: Union[PyGraph, nx.Graph]) -> None:
```

Comment:
Probably this type hint should be a string since `nx.Graph` is not defined if users don't have it installed..

Reply:
Maybe take a look at how #535 deals with that. In the `PyQuanteDriver` something similar to this may occur. Maybe @manoelmarques can also tell you directly :+1:

Reply:
Yes, strings like it is already done for the `Matplotlib` type hints in `LatticeDrawStyle`